### PR TITLE
Only use for_end_of_month if present; add sentry context

### PR DESCRIPTION
**Pull request by rsanheim**
It appears we are getting some cases where we have an empty end_of_month
params..this will stop trying to parse those and also add some extra
sentry context to help debugging.

**Story card:** [ch4457](https://app.clubhouse.io/simpledotorg/story/4457/argumenterror-invalid-date-for-drugstock-controller)

Sentry issue: https://sentry.io/organizations/resolve-to-save-lives/issues/2561940349/?referrer=Clubhouse

### CODE REVIEW

File: app/controllers/concerns/set_for_end_of_month.rb [hunk 1]
```diff
@@ -3,7 +3,17 @@ def set_for_end_of_month
     # Defaults to current month if it's the last week of the month.
     @show_current_month = ((Date.current.end_of_month - Date.current).to_i <= 7)
 
-    @for_end_of_month ||= if params[:for_end_of_month]
+    Sentry.configure_scope do |scope|
```
Comment (harimohanraj89) [inline app/controllers/concerns/set_for_end_of_month.rb:6]: This is set for the scope of this request cycle?

File: app/controllers/concerns/set_for_end_of_month.rb [hunk 2]
```diff
@@ -3,7 +3,17 @@ def set_for_end_of_month
...
+      user = @current_admin || @current_user
```
Comment (harimohanraj89) [inline app/controllers/concerns/set_for_end_of_month.rb:7]: Is it easy for us to pull this out and apply it to every controller action under `AdminController`?

**Review by harimohanraj89 — COMMENTED** [review]
Are we adding the Sentry stuff because we're not sure if a blank param is the problem?

File: app/controllers/concerns/set_for_end_of_month.rb [hunk 3]
```diff
@@ -3,7 +3,17 @@ def set_for_end_of_month
...
+      scope.set_user(id: user.id)
```
Comment (prabhanshuguptagit) [inline app/controllers/concerns/set_for_end_of_month.rb:8]: It seems like we don't have the user id for any dashboard controllers. Shall we just add it at `AdminController`?

File: app/controllers/concerns/set_for_end_of_month.rb [hunk 4]
```diff
@@ -3,7 +3,17 @@ def set_for_end_of_month
     # Defaults to current month if it's the last week of the month.
```
Comment (prabhanshuguptagit) [inline app/controllers/concerns/set_for_end_of_month.rb:3]: The `.present?` seems to fix the issue, why are we adding the Sentry information?

Comment (rsanheim) [reply app/controllers/concerns/set_for_end_of_month.rb:3]: I'm not sure that is the issue here...I want to have more context if this error (or others) show up in Sentry.

File: app/controllers/concerns/set_for_end_of_month.rb (hunk 1 above)
Comment (rsanheim) [reply app/controllers/concerns/set_for_end_of_month.rb:6]: Yup, I believe so.  The sentry API is confusing now about this, but this should be for the length of the request.

File: app/controllers/concerns/set_for_end_of_month.rb (hunk 2 above)
Comment (rsanheim) [reply app/controllers/concerns/set_for_end_of_month.rb:7]: Not easy enough where I want to do it in this PR 😁 ...this module gets mixed into Admin and API controllers, so I want to keep this change isolated.

Comment (harimohanraj89) [reply app/controllers/concerns/set_for_end_of_month.rb:7]: Fair enough 😅

**Review by harimohanraj89 — APPROVED** [review]
Coolcool let's learn about this error 🚀

**Pull request merged**